scripts/check-docs.py

```
@@ -0,0 +1,413 @@
+    return loader.construct_mapping(node, deep=True)
+
+
+_NavLoader.add_multi_constructor("", _ignore_unknown_tag)
+
+
+def load_nav_paths() -> set[str]:
+    """Every docs-relative page path reachable from the nav tree."""
+    with MKDOCS_YML.open(encoding="utf-8") as handle:
+        # _NavLoader is a SafeLoader subclass; see its docstring for why this is
+        # not the yaml.load foot-gun it looks like.
+        config = yaml.load(handle, Loader=_NavLoader)
```

Ruff (0.16.0)

[error] Probable use of unsafe loader `_NavLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)